### PR TITLE
docs: update license and migrate docs to mkdocs

### PR DESCRIPTION
**Pull request by gcroci2**
The new docs are visible [here](https://neurogym.github.io/neurogym/latest/). Some notes:

- As a suggestion to review this PR, you can just avoid looking at the docs/ files (the majority of the edits), and instead look at the appearance of the docs on the website. Just in case you have comments dig into the files. As I said during the meeting, I didn't focus on editing the old docs, and I wouldn't do it now. Let's focus on the most important details only (at least for now). 
- I had to exclude paths from the linkspector. About it, I really don't like that a bot gets assigned as a reviewer and adds comments on the PR. It can get easily very verbose, and I think something like [what we had before](https://github.com/ANNUBS/annubes/blob/main/.github/workflows/markdown-links.yml) was much less disrupting. Any major reason for switching to linkspector?
- Something went wrong [in the last merge you did](https://github.com/neurogym/neurogym/blob/460aeab2fcf60ed8fd09bc47966376022dacb3a9/README.md) @DaniBodor (see the original readme on this branch's readme), but now it should be fixed. Just double-check that nothing else went wrong there.

### CODE REVIEW

**Comment by gcroci2** [comment]
Hi @manuelmolano and @nacloos,

I think it would be helpful if you could also review and give the okay for this PR.

- Manual, Nathan: There's no need to check the code; please just take a look at the new NeuroGym documentation website: https://neurogym.github.io/neurogym/. I’ve tried to keep the docs as consistent as possible with the previous version.
- Note: I removed the video.py file since I couldn’t find where (or if) it was being rendered.

If you both approve, I plan to archive the [neurogym.github.io](https://github.com/neurogym/neurogym.github.io) repository, which was created for the older version of the docs. Now that everything is hosted on a dedicated branch ([gh-pages](https://github.com/neurogym/neurogym/tree/gh-pages)) within the main NeuroGym repo, the separate repository is no longer necessary (and arguably wasn’t needed initially).

File: mkdocs.yml
Comment (DaniBodor) [inline mkdocs.yml:4]: duplicated below

```suggestion
```

File: README.dev.md
Comment (DaniBodor) [inline README.dev.md:103]: Doesn't mike automatically deploy new docs upon each release?
If not, In would suggest adding a job to the release workflow so that it does happen (in a separate PR).

File: README.dev.md
Comment (DaniBodor) [inline README.dev.md:74]: I would leave this out. It's part of the developer dependencies, and the website shows the instructions of how to run it (which is intuitive anyway).
Also, should we move the mypy settings into a separate .mypy.toml file instead?

```suggestion
We use [Mypy](http://mypy-lang.org/) as static type checker and the configurations are set in [pyproject.toml](pyproject.toml) file.
```

File: README.dev.md
Comment (DaniBodor) [inline README.dev.md:107]: This is not really clear to me.
Also, do we need this info here? Isn't this a bit of detailed usage instructions for an external package? (same for previous paragraph about deleting changes)?

**Review by DaniBodor — APPROVED** [review]
Really nice documentation page :)
I think the mouse as the icon makes less sense here than it did in annubes, but still cute :)

apart from that I only left a few small comments

>   * As a suggestion to review this PR, you can just avoid looking at the docs/ files (the majority of the edits), and instead look at the appearance of the docs on the website. Just in case you have comments dig into the files. As I said during the meeting, I didn't focus on editing the old docs, and I wouldn't do it now. Let's focus on the most important details only (at least for now).

I didnt look at the contents at all. Opening the pages and navigating through looks good!

>   * I had to exclude paths from the linkspector. About it, I really don't like that a bot gets assigned as a reviewer and adds comments on the PR. It can get easily very verbose, and I think something like [what we had before](https://github.com/ANNUBS/annubes/blob/main/.github/workflows/markdown-links.yml) was much less disrupting. Any major reason for switching to linkspector?

I agree about the bot reviweer. I imagine it is a setting one can turn off. 
The reason I migrated is because I was getting bugs in the old system and when troubleshooting I saw that the old system is deprecated and the same developer created linkspector to take over. I didnt put much though into it, just migrated and it worked without bugs.
We can either move back to the previous one and resolve the bugs, or look into turning off the bot reviewer for linkspector. I open #96 for that and will look into it.

>   * Something went wrong [in the last merge you did](https://github.com/neurogym/neurogym/blob/460aeab2fcf60ed8fd09bc47966376022dacb3a9/README.md) @DaniBodor (see the original readme on this branch's readme), but now it should be fixed. Just double-check that nothing else went wrong there.

Not sure how that happened, thanks for solving it!

**Comment by gcroci2** [comment]
> Really nice documentation page :) I think the mouse as the icon makes less sense here than it did in annubes, but still cute :)

Would a brain fit better? @DaniBodor 

image

> The reason I migrated is because I was getting bugs in the old system and when troubleshooting I saw that the old system is deprecated and the same developer created linkspector to take over. I didnt put much though into it, just migrated and it worked without bugs.
We can either move back to the previous one and resolve the bugs, or look into turning off the bot reviewer for linkspector. I open https://github.com/neurogym/neurogym/issues/96 for that and will look into it.

Definitely let's keep the new one then :)

File: README.dev.md
Comment (gcroci2) [reply README.dev.md:103]: It doesn't; I opened a new issue for that (#100, we can pick it up after the holidays, or feel free to pick it up before then - I won't have the time). We will do the first release independently from this new issue.

File: README.dev.md
Comment (DaniBodor) [reply README.dev.md:97]: I understand your reasons for having them here then. I still don't understand what you mean in line 107.

**Comment by DaniBodor** [comment]
> > Really nice documentation page :) I think the mouse as the icon makes less sense here than it did in annubes, but still cute :)
> 
> Would a brain fit better? @DaniBodor
> image

haha, cool. Yeah, I think either of those fits nicely.


> > The reason I migrated is because I was getting bugs in the old system and when troubleshooting I saw that the old system is deprecated and the same developer created linkspector to take over. I didnt put much though into it, just migrated and it worked without bugs.
> > We can either move back to the previous one and resolve the bugs, or look into turning off the bot reviewer for linkspector. I open #96 for that and will look into it.
> 
> Definitely let's keep the new one then :)

+1

> From one of your comments:
> 
> > Also, should we move the mypy settings into a separate .mypy.toml file instead?
> 
> Why is a separate file with a few settings beneficial rather than having all in one? For the ruff settings it makes sense because they're quite a lot, but for mypy I don't think it is needed.

good point. I was indeed thinking along the lines of ruff and bump-my-version. But given that its a single group, I agree they might as well stay here.

File: README.dev.md
Comment (gcroci2) [reply README.dev.md:97]: Would this be clearer?

Use `mike serve` to review all versions of the site that have been committed to the `gh-pages` branch. This command is best for verifying the production website as it exists in the repository.

If you have uncommitted changes that you want to preview, use `mkdocs serve` instead of `mike serve`. This will allow you to view your changes locally without committing them first.

**Comment by gcroci2** [comment]
Last minute change, I fixed the license issue here directly @DaniBodor and linked to that issue as well. I double checked with Patrick that this was fine.

**Comment by DaniBodor** [comment]
> Last minute change, I fixed the license issue here directly @DaniBodor and linked to that issue as well. I double-checked with Patrick that this is fine.

I would not do that here. It kind of hides this change, which I think is significant, under an unrelated issue.
Also, there is more to it than just swapping out the license files:
- it needs to be clearly documented until which version the old license applies and from which version the new one does, which is missing here.
- a few other files (citation.cff, pyproject.toml, NOTICE) explicitly refer to the MIT license, which need to be updated as well.

Finally, by putting it in a separate PR, we can also more easily tag former contributors in the issue to make them aware of the changed license.

**Comment by gcroci2** [comment]
> I would not do that here. It kind of hides this change, which I think is significant, under an unrelated issue. Also, there is more to it than just swapping out the license files:

I edited the title accordingly and I linked the license issue here as well. I think that now it is pretty clear.

> * it needs to be clearly documented until which version the old license applies and from which version the new one does, which is missing here.

It will apply to all the commits on the branch with the Apache license from now on (now this branch, later the dev, later the main). According to Patrick we don't really need to do anything else, nor adding disclaimers around. I explicitly asked to avoid keep discussing on this.

**Comment by gcroci2** [comment]
Please note that I've updated [the comment](https://github.com/neurogym/neurogym/pull/93#issuecomment-2550813140) addressed to you @manuelmolano @nacloos, so If you haven't read it yet no action is needed, otherwise please read it again :)

You can safely skip all the rest of the comments/reviews in this pull request.

**Comment by DaniBodor** [comment]
> It will apply to all the commits on the branch with the Apache license from now on (now this branch, later the dev, later the main). According to Patrick we don't really need to do anything else, nor adding disclaimers around (which I did add at the beginning but I now removed after chatting with him). I explicitly asked to avoid keep discussing on this.

I really disagree with this. It's still hidden, it's scope creep (this PR has nothing to do with licenses), the fact that it was a last minute addition also kinda implies that it doesnt relate to this PR. What is the argument in favor of adding it here if not specifically to kind of shove it under the rug?

We are already shortcutting the process by not getting approval from all contributors. Everything I read everywhere specifies that this is a requirement for relicensing projects. 
Also, what would be the argument not to be explicit about the change in license. Any piece of the code that has not been changed since that date is still under the old license. It's not so much about future developer's making changes to this code base, but about people picking stuff up from this code base and moving it somewhere else. If they do due diligence and mention our license in that case, they would need to dig very deep to find that the license changed. Why not help them out by making it clear from the start that code as per date X is under one license and older code is under another?

**Comment by gcroci2** [comment]
@DaniBodor I've squashed all the changes relative to the license in this commit (https://github.com/neurogym/neurogym/pull/93/commits/c0f6cdc12b7ecab37ff12eab36afa7b2e49a6f75); I could add a tag to the commit to make the change of license even more evident and back-traceable. Alternatively, you can open a new PR, I can revert this last commit, and we keep on hold the license issue (but in the meantime we merge this and I release to PyPI anyway).

File: README.dev.md
Comment (DaniBodor) [reply README.dev.md:97]: yes!

**Comment by gcroci2** [comment]
Just as a reference, the license change is now tagged ([v0.1.1-license-apache-2.0](https://github.com/neurogym/neurogym/releases/tag/v0.1.1-license-apache-2.0))

**Pull request merged**
